src/Microsoft.TemplateEngine.Core.Contracts/IProcessorState.cs

```diff
@@ -31,5 +32,7 @@ public interface IProcessorState
         void SeekBackUntil(ITokenTrie match, bool consume);
 
         void SeekBackWhile(ITokenTrie match);
+
+        void Inject(Stream staged, int sequenceNumberEffect);
```

Comment:
I believe this is what is going on:
Using Inject() is logically similar to seeking backwards in a stream - but we use forward only streams. 

The specific use in this PR is:
- The InlineMarkupConditional read some data to process.
- After examining the data, InlineMarkupConditional decides it shouldn't be processing that data.
- Prior to this PR, the unprocessed data would be written to the output stream, and processing of the input data would continue after that data.
- With this PR, the unprocessed source data is "injected" back into the source stream, so it can be read again, and processed by other operations, if applicable.